app/grants/views.py

```diff
@@ -80,61 +80,36 @@ def grant_details(request, grant_id, grant_slug):
-            subscription_terminated(grant, sub)
-
-    # TODO: Determine how we want to chunk out articles and where we want to store this data.
-    activity_data = [{
-        'title': 'allow funder to turn off auto approvals during bounty creation',
-        'date': '08.02.2018',
-        'description':
-            'Vestibulum ante ipsum primis in faucibus orci luctus ultrices '
-            'posuere cubilia Curae; Proin vel ante.',
-    }, {
-        'title': 'Beyond The Naked Eye',
-        'date': '2012 - Present',
-        'description':
-            'What is the loop of Creation? How is there something from nothing? '
-            'In spite of the fact that it is impossible to prove that anything exists beyond '
-            'one’s perception since any such proof would involve one’s perception (I observed it, '
-            'I heard it, I thought about it, I calculated it, and etc.), science deals with a '
-            'so-called objective reality “out there,” beyond one’s perception professing to '
-            'describe Nature objectively (as if there was a Nature or reality external '
-            'to one’s perception). The shocking impact of Matrix was precisely the valid '
-            'possibility that what we believed to be reality was but our perception; however, '
-            'this was presented through showing a real reality wherein the perceived reality was a '
-            'computer simulation. Many who toy with the idea that perhaps, indeed, we are computer '
-            'simulations, deviate towards questions, such as, who could create such software and what '
-            'kind of hardware would be needed for such a feat. Although such questions assume that reality '
-            'is our perception, they also axiomatically presuppose the existence of an objective '
-            'deterministic world “out there” that nevertheless must be responsible for how we perceive '
-            'our reality. This is a major mistake emphasizing technology and algorithms instead of trying '
-            'to discover the nature of reality and the structure of creation. As will be shown in the following, '
-            'the required paradigm shift from “perception is our reality fixed within an objective world,” '
-            'to “perception is reality without the need of an objective world ‘out there,” '
-            'is provided by a dynamic logical structure. The Holophanic loop logic is responsible '
-            'for a consistent and complete worldview that not only describes, but also creates whatever '
-            'can be perceived or experienced.'
-    }, {
-        'title': 'Awesome Update',
-        'date': '08.02.2018',
-        'description': 'Some awesome update about this project.',
-    }, {
-        'title': 'Stellar Update',
-        'date': '08.02.2018',
-        'description': 'Another stellar update about this project.',
-    }]
+        print(request.POST)
+        if 'contract_address' in request.POST:
```

Comment:
This shouldn't be the latest revision to grants..? I know I've fixed the ability for *anyone* to no longer cancel a grant...

Reply:
Yes, only an admin is shown the "Cancel This Grant" button. I have added here a series of if statements, based on the names of inputs in the different forms, to handle the different form submissions on the grants/details page. Currently, there is a form for the "Cancel This Grant" button, the updates form, and the edit details form.

Reply:
@captnseagraves I'm saying in the actual backend here, *anyone* can simply `POST` and delete it.

Reply:
@mbeacom Gotcha, makes sense. Unfortunately, I have not seen your implementation for that. I pulled the latest grants and added to the `if request.method == 'POST':` statement for each form.  Would the solution look something like an additional `if` statement, `if profile == grant.admin_profile:`

Reply:
hmmm... No that wouldn't work. Can you point me in the right direction?